### PR TITLE
fix(tianmu): fix STR_TO_DATE() Function does not work well. (#975)

### PR DESCRIPTION
**Pull request by lujiashun**
## Summary about this PR

Issue Number: ref #975

1 add MicroSecond logic;
2 allow "00-00-00 11:11:11" datime format to follow innodb behavior;
3 refactor code,change CT to ColumnType;

## Tests Check List

- [ ] Unit test
- [X] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

## Changelog

- [x] New Feature
- [ ] Bug Fix
- [ ] Performance Improvement
- [ ] Build/Testing/CI/CD
- [ ] Documentation
- [ ] Not for changelog (changelog entry is not required)

## Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features

### CODE REVIEW

File: storage/tianmu/types/value_parser4txt.cpp [hunk 1]
```diff
@@ -662,7 +662,8 @@ common::ErrorCode ValueParserForText::ParseDateTimeOrTimestamp(const BString &ti
   }
 
   if (tianmu_err_code != common::ErrorCode::OUT_OF_RANGE && buflen != 0)
-    year = TianmuDateTime::ToCorrectYear((uint)year, at);
+    if (year != kTianmuDateSpec.Year())
```
Comment (RingsC) [inline storage/tianmu/types/value_parser4txt.cpp]:
```
if (a)
if (b)
xxx
```
can be combine into 
```
if (a && b)
  xxx
```

File: storage/tianmu/types/value_parser4txt.cpp [hunk 2]
```diff
@@ -743,10 +746,22 @@ common::ErrorCode ValueParserForText::ParseDateTimeOrTimestamp(const BString &ti
...
   if (tianmu_err_code == common::ErrorCode::FAILED)
     tianmu_err_code = common::ErrorCode::OUT_OF_RANGE;
 
-  EatWhiteSigns(buf, buflen);
+  if (system::EatDTSeparators(buf, buflen))  // contains micro seconds
+  {
+    tianmu_err_code = system::EatUInt(buf, buflen, microsecond);
+    if (!common::IsError(tianmu_err_code)) {
+      if (!TianmuDateTime::CanBeMicroSecond(microsecond)) {
+        rcv = TianmuDateTime((short)year, (short)month, (short)day, (short)hour, (short)minute, (short)second,
```
Comment (RingsC) [inline storage/tianmu/types/value_parser4txt.cpp]: using c++ style type casting.

File: storage/tianmu/types/tianmu_data_types.h [hunk 3]
```diff
@@ -301,6 +301,8 @@ class TianmuDateTime : public ValueBasic<TianmuDateTime> {
   TianmuDateTime(int64_t dt, common::CT at);
```
Comment (RingsC) [inline storage/tianmu/types/tianmu_data_types.h]: `CT` can be `ColumnType`, it can be in an other PR.

**Review by RingsC — CHANGES_REQUESTED** [review]

File: storage/tianmu/types/tianmu_data_types.h (hunk 3 above)
Comment (lujiashun) [reply storage/tianmu/types/tianmu_data_types.h]: Fixed it.

File: storage/tianmu/types/value_parser4txt.cpp (hunk 2 above)
Comment (lujiashun) [reply storage/tianmu/types/value_parser4txt.cpp]: Fixed it.

File: storage/tianmu/types/value_parser4txt.cpp (hunk 1 above)
Comment (lujiashun) [reply storage/tianmu/types/value_parser4txt.cpp]: Fixed it.

**Review by RingsC — APPROVED** [review]
LGTM

**Review by duanfuxiang0 — APPROVED** [review]
LGTM

**Pull request merged**